metashare/api/jobs.py

```diff
@@ -470,3 +471,39 @@ def populate_github_users(repository):
 
 
 populate_github_users_job = job(populate_github_users)
+
+
+def submit_review(*, user, task, data):
+    try:
+        task.refresh_from_db()
+        repository = get_repo_info(user, repo_id=task.project.repository.repo_id)
+        pr = repository.pull_request(task.pr_number)
+        # We always COMMENT so as not to change the PR's status:
+        pr.create_review(data["notes"], event="COMMENT")
+        if task.commits:
+            state_for_status = {
+                # "": "pending",
+                # "": "error",
+                TASK_REVIEW_STATUS.Approved: "success",
+                TASK_REVIEW_STATUS["Changes requested"]: "failure",
+            }.get(data["status"])
+            repository.create_status(
+                task.commits[0],
+                state_for_status,
+                description=data["notes"],
+                context="MetaShare Review",
```

Comment:
I kinda think we might as well include a `target_url` too, to link back to the relevant URL on MetaShare.

Reply:
@wlonk I think we can pull out a helper to construct user-facing URLs on the back end, from what's currently there in `alert_user_about_expiring_orgs`: https://github.com/oddbird/MetaShare/blob/master/metashare/api/jobs.py#L99-L108

Something .e.g.

```py
def get_url(path):
    domain = Site.objects.first().domain
    should_be_http = not settings.SECURE_SSL_REDIRECT or domain.startswith(
        "localhost"
    )
    scheme = "http" if should_be_http else "https"
    return (
        furl(f"{scheme}://{domain}")
        .set(path=path)
        .url
    )
```

```py
# in alert_user_about_expiring_org
metashare_link = get_url(path=["repositories", repo.slug, project.slug, task.slug])

# in submit_review
target_url = get_url(path=["repositories", task.project.repository.slug, task.project.slug, task.slug])
```

Reply:
@jgerigmeyer  or @wlonk : Question, is this link visible in MS or for use on Github?

Reply:
This is on GH -- the "Details" link back to MS from the commit status, like:

![Screenshot 2020-02-17 13 35 30](https://user-images.githubusercontent.com/552316/74678962-79431e80-518a-11ea-8bb6-af6a50cfd9fc.png)

Reply:
Sweet :+1:

Reply:
Addressed in https://github.com/oddbird/MetaShare/pull/164/commits/d439639bbbb248d6c46c53360e70ffdfd3a2967b